### DISCUSSION TITLE
Textadept for 64-bit arm Linux (e.g. for Termux, Raspberry Pi, etc.)

### DISCUSSION THREAD

**Discussion by kumoshk** [orbitalquark/textadept#133] [General]
I tried compiling Textadept on Termux (a command-line Linux for Android). When I did `build deps`, I got the following error:

2021-09-18 17:53:11 (519 KB/s) - ‘cdk-5.0-20200923.tgz’ saved [545093/545093]

if [ -d cdk ]; then rm -r cdk; fi
mkdir cdk && tar xzf cdk-5.0-20200923.tgz -C cdk && mv cdk/*/* cdk
mv cdk/include/*.h cdk
cdk/gen-scale.sh SLIDER Slider Int int cdk/gen-slider.h > cdk/slider.h
/data/data/com.termux/files/usr/bin/sh: 1: cdk/gen-scale.sh: Permission denied
make: *** [Makefile:405: cdk] Error 126

**Comment by orbitalquark** [comment]
In *src/Makefile* on line 405 and line 406, try prepending "sh " to the lines so they read "sh $@/gen-scale.sh [...]". I'm guessing executable permissions were not preserved when untarring the cdk tgz on your platform.

**Reply by kumoshk** [reply]
Thanks! I'll have to try that out.

**Reply by kumoshk** [reply]
Okay, I tried it, and that worked for `make deps`, but when I did `make curses` on Termux, I get this output:

make curses
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall scintilla/src/AutoComplete.cxx -o AutoComplete.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall scintilla/src/CallTip.cxx -o CallTip.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall scintilla/src/CaseConvert.cxx -o CaseConvert.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall scintilla/src/CaseFolder.cxx -o CaseFolder.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall scintilla/src/CellBuffer.cxx -o CellBuffer.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall scintilla/src/CharacterCategoryMap.cxx -o CharacterCategoryMap.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall scintilla/src/CharacterType.cxx -o CharacterType.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall scintilla/src/CharClassify.cxx -o CharClassify.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall scintilla/src/ContractionState.cxx -o ContractionState.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall scintilla/src/DBCS.cxx -o DBCS.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall scintilla/src/Decoration.cxx -o Decoration.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall scintilla/src/Document.cxx -o Document.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall scintilla/src/EditModel.cxx -o EditModel.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall scintilla/src/Editor.cxx -o Editor.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall scintilla/src/EditView.cxx -o EditView.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall scintilla/src/Geometry.cxx -o Geometry.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall scintilla/src/Indicator.cxx -o Indicator.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall scintilla/src/KeyMap.cxx -o KeyMap.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall scintilla/src/LineMarker.cxx -o LineMarker.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall scintilla/src/MarginView.cxx -o MarginView.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall scintilla/src/PerLine.cxx -o PerLine.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall scintilla/src/PositionCache.cxx -o PositionCache.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall scintilla/src/RESearch.cxx -o RESearch.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall scintilla/src/RunStyles.cxx -o RunStyles.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall scintilla/src/ScintillaBase.cxx -o ScintillaBase.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall scintilla/src/Selection.cxx -o Selection.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall scintilla/src/Style.cxx -o Style.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall scintilla/src/UniConversion.cxx -o UniConversion.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall scintilla/src/UniqueString.cxx -o UniqueString.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall scintilla/src/ViewStyle.cxx -o ViewStyle.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall scintilla/src/XPM.cxx -o XPM.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall lexilla/lexlib/Accessor.cxx -o Accessor.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall lexilla/lexlib/CharacterCategory.cxx -o CharacterCategory.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall lexilla/lexlib/CharacterSet.cxx -o CharacterSet.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall lexilla/lexlib/DefaultLexer.cxx -o DefaultLexer.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall lexilla/lexlib/LexerBase.cxx -o LexerBase.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall lexilla/lexlib/LexerModule.cxx -o LexerModule.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall lexilla/lexlib/LexerNoExceptions.cxx -o LexerNoExceptions.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall lexilla/lexlib/LexerSimple.cxx -o LexerSimple.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall lexilla/lexlib/PropSetSimple.cxx -o PropSetSimple.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall lexilla/lexlib/StyleContext.cxx -o StyleContext.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall lexilla/lexlib/WordList.cxx -o WordList.o
g++ -c -Os -std=c++17 -DLUA_USE_LINUX -Ilua/src -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall -DCURSES -D_XOPEN_SOURCE=600 -I/data/data/com.termux/files/usr/include -DNO_SCITE -DNO_DLL LexLPeg.cxx -o LexLPeg-curses.o
g++ -c -Os -std=c++17 -pedantic -DCURSES -DSCI_LEXER -DNDEBUG -Iscintilla/include -Iscintilla/src -Ilexilla/include -Ilexilla/lexlib -Wall -D_XOPEN_SOURCE=600 -I/data/data/com.termux/files/usr/include scintilla/curses/ScintillaCurses.cxx -o ScintillaCurses.o
gcc -c -Os -DLUA_USE_LINUX -Ilua/src -std=c99 -pedantic -DCURSES -Iscintilla/include -Igtdialog -Wall -Iscintilla/curses -Itermkey -Icdk -D_XOPEN_SOURCE=600 -I/data/data/com.termux/files/usr/include textadept.c -o textadept-curses.o
textadept.c:4:9: warning: '_XOPEN_SOURCE' macro redefined [-Wmacro-redefined]
#define _XOPEN_SOURCE 500 // for readlink from unistd.h
        ^
<command line>:3:9: note: previous definition is here
#define _XOPEN_SOURCE 600
        ^
textadept.c:188:31: warning: unused variable 'tab_sync' [-Wunused-variable]
static bool initing, closing, tab_sync, dialog_active;
                              ^
textadept.c:1453:12: warning: unused function 'timed_out' [-Wunused-function]
static int timed_out(void *args) {
           ^
3 warnings generated.
gcc -c -Os -DLUA_USE_LINUX -Ilua/src -ULUA_LIB lua/src/lapi.c -o lapi.o
gcc -c -Os -DLUA_USE_LINUX -Ilua/src -ULUA_LIB lua/src/lcode.c -o lcode.o
gcc -c -Os -DLUA_USE_LINUX -Ilua/src -ULUA_LIB lua/src/lctype.c -o lctype.o
gcc -c -Os -DLUA_USE_LINUX -Ilua/src -ULUA_LIB lua/src/ldebug.c -o ldebug.o
gcc -c -Os -DLUA_USE_LINUX -Ilua/src -ULUA_LIB lua/src/ldo.c -o ldo.o
gcc -c -Os -DLUA_USE_LINUX -Ilua/src -ULUA_LIB lua/src/ldump.c -o ldump.o
gcc -c -Os -DLUA_USE_LINUX -Ilua/src -ULUA_LIB lua/src/lfunc.c -o lfunc.o
gcc -c -Os -DLUA_USE_LINUX -Ilua/src -ULUA_LIB lua/src/lgc.c -o lgc.o
gcc -c -Os -DLUA_USE_LINUX -Ilua/src -ULUA_LIB lua/src/linit.c -o linit.o
gcc -c -Os -DLUA_USE_LINUX -Ilua/src -ULUA_LIB lua/src/llex.c -o llex.o
gcc -c -Os -DLUA_USE_LINUX -Ilua/src -ULUA_LIB lua/src/lmem.c -o lmem.o
gcc -c -Os -DLUA_USE_LINUX -Ilua/src -ULUA_LIB lua/src/lobject.c -o lobject.o
gcc -c -Os -DLUA_USE_LINUX -Ilua/src -ULUA_LIB lua/src/lopcodes.c -o lopcodes.o
gcc -c -Os -DLUA_USE_LINUX -Ilua/src -ULUA_LIB lua/src/lparser.c -o lparser.o
gcc -c -Os -DLUA_USE_LINUX -Ilua/src -ULUA_LIB lua/src/lstate.c -o lstate.o
gcc -c -Os -DLUA_USE_LINUX -Ilua/src -ULUA_LIB lua/src/lstring.c -o lstring.o
gcc -c -Os -DLUA_USE_LINUX -Ilua/src -ULUA_LIB lua/src/ltable.c -o ltable.o
gcc -c -Os -DLUA_USE_LINUX -Ilua/src -ULUA_LIB lua/src/ltm.c -o ltm.o
gcc -c -Os -DLUA_USE_LINUX -Ilua/src -ULUA_LIB lua/src/lundump.c -o lundump.o
gcc -c -Os -DLUA_USE_LINUX -Ilua/src -ULUA_LIB lua/src/lvm.c -o lvm.o
gcc -c -Os -DLUA_USE_LINUX -Ilua/src -ULUA_LIB lua/src/lzio.c -o lzio.o
gcc -c -Os -DLUA_USE_LINUX -Ilua/src -ULUA_LIB lua/src/lauxlib.c -o lauxlib.o
gcc -c -Os -DLUA_USE_LINUX -Ilua/src -ULUA_LIB lua/src/lbaselib.c -o lbaselib.o
gcc -c -Os -DLUA_USE_LINUX -Ilua/src -ULUA_LIB lua/src/lcorolib.c -o lcorolib.o
gcc -c -Os -DLUA_USE_LINUX -Ilua/src -ULUA_LIB lua/src/ldblib.c -o ldblib.o
gcc -c -Os -DLUA_USE_LINUX -Ilua/src -ULUA_LIB lua/src/liolib.c -o liolib.o
gcc -c -Os -DLUA_USE_LINUX -Ilua/src -ULUA_LIB lua/src/lmathlib.c -o lmathlib.o
gcc -c -Os -DLUA_USE_LINUX -Ilua/src -ULUA_LIB lua/src/loadlib.c -o loadlib.o
gcc -c -Os -DLUA_USE_LINUX -Ilua/src -ULUA_LIB lua/src/lstrlib.c -o lstrlib.o
gcc -c -Os -DLUA_USE_LINUX -Ilua/src -ULUA_LIB lua/src/ltablib.c -o ltablib.o
gcc -c -Os -DLUA_USE_LINUX -Ilua/src -ULUA_LIB lua/src/lutf8lib.c -o lutf8lib.o
gcc -c -Os -DLUA_USE_LINUX -Ilua/src -ULUA_LIB -std=c99 -pedantic lua/src/loslib.c -o loslib-curses.o
gcc -c -Os -DLUA_USE_LINUX -Ilua/src lua/src/lib/lpcap.c -o lpcap.o
gcc -c -Os -DLUA_USE_LINUX -Ilua/src lua/src/lib/lpcode.c -o lpcode.o
gcc -c -Os -DLUA_USE_LINUX -Ilua/src lua/src/lib/lpprint.c -o lpprint.o
gcc -c -Os -DLUA_USE_LINUX -Ilua/src lua/src/lib/lptree.c -o lptree.o
gcc -c -Os -DLUA_USE_LINUX -Ilua/src lua/src/lib/lpvm.c -o lpvm.o
gcc -c -Os -DLUA_USE_LINUX -Ilua/src lua/src/lib/lfs.c -o lfs.o
gcc -c -Os -std=c99 -pedantic -DCURSES -DNOHELP -DLIBRARY -I. -Icdk -D_XOPEN_SOURCE=600 -I/data/data/com.termux/files/usr/include \
        gtdialog/gtdialog.c -o gtdialog-curses.o
gcc -c -Os -std=c99 -D_XOPEN_SOURCE=600 -I/data/data/com.termux/files/usr/include termkey/termkey.c -o termkey.o
gcc -c -Os -std=c99 -D_XOPEN_SOURCE=600 -I/data/data/com.termux/files/usr/include termkey/driver-ti.c -o driver-ti.o
gcc -c -Os -std=c99 -D_XOPEN_SOURCE=600 -I/data/data/com.termux/files/usr/include termkey/driver-csi.c -o driver-csi.o
gcc -c -Os -D_GNU_SOURCE -Itermkey -Icdk -D_XOPEN_SOURCE=600 -I/data/data/com.termux/files/usr/include cdk/binding.c -o cdk-binding.o
gcc -c -Os -D_GNU_SOURCE -Itermkey -Icdk -D_XOPEN_SOURCE=600 -I/data/data/com.termux/files/usr/include cdk/buttonbox.c -o cdk-buttonbox.o
gcc -c -Os -D_GNU_SOURCE -Itermkey -Icdk -D_XOPEN_SOURCE=600 -I/data/data/com.termux/files/usr/include cdk/cdk.c -o cdk-cdk.o
gcc -c -Os -D_GNU_SOURCE -Itermkey -Icdk -D_XOPEN_SOURCE=600 -I/data/data/com.termux/files/usr/include cdk/cdk_display.c -o cdk-cdk_display.o
gcc -c -Os -D_GNU_SOURCE -Itermkey -Icdk -D_XOPEN_SOURCE=600 -I/data/data/com.termux/files/usr/include cdk/cdk_objs.c -o cdk-cdk_objs.o
gcc -c -Os -D_GNU_SOURCE -Itermkey -Icdk -D_XOPEN_SOURCE=600 -I/data/data/com.termux/files/usr/include cdk/cdkscreen.c -o cdk-cdkscreen.o
gcc -c -Os -D_GNU_SOURCE -Itermkey -Icdk -D_XOPEN_SOURCE=600 -I/data/data/com.termux/files/usr/include cdk/draw.c -o cdk-draw.o
gcc -c -Os -D_GNU_SOURCE -Itermkey -Icdk -D_XOPEN_SOURCE=600 -I/data/data/com.termux/files/usr/include cdk/entry.c -o cdk-entry.o
gcc -c -Os -D_GNU_SOURCE -Itermkey -Icdk -D_XOPEN_SOURCE=600 -I/data/data/com.termux/files/usr/include cdk/fselect.c -o cdk-fselect.o
gcc -c -Os -D_GNU_SOURCE -Itermkey -Icdk -D_XOPEN_SOURCE=600 -I/data/data/com.termux/files/usr/include cdk/itemlist.c -o cdk-itemlist.o
gcc -c -Os -D_GNU_SOURCE -Itermkey -Icdk -D_XOPEN_SOURCE=600 -I/data/data/com.termux/files/usr/include cdk/label.c -o cdk-label.o
gcc -c -Os -D_GNU_SOURCE -Itermkey -Icdk -D_XOPEN_SOURCE=600 -I/data/data/com.termux/files/usr/include cdk/mentry.c -o cdk-mentry.o
cdk/mentry.c:705:16: warning: result of comparison of constant -1 with expression of type 'char' is always false [-Wtautological-constant-out-of-range-compare]
   if (newchar == ERR)
       ~~~~~~~ ^  ~~~
1 warning generated.
gcc -c -Os -D_GNU_SOURCE -Itermkey -Icdk -D_XOPEN_SOURCE=600 -I/data/data/com.termux/files/usr/include cdk/popup_label.c -o cdk-popup_label.o
gcc -c -Os -D_GNU_SOURCE -Itermkey -Icdk -D_XOPEN_SOURCE=600 -I/data/data/com.termux/files/usr/include cdk/scroll.c -o cdk-scroll.o
gcc -c -Os -D_GNU_SOURCE -Itermkey -Icdk -D_XOPEN_SOURCE=600 -I/data/data/com.termux/files/usr/include cdk/scroller.c -o cdk-scroller.o
gcc -c -Os -D_GNU_SOURCE -Itermkey -Icdk -D_XOPEN_SOURCE=600 -I/data/data/com.termux/files/usr/include cdk/select_file.c -o cdk-select_file.o
gcc -c -Os -D_GNU_SOURCE -Itermkey -Icdk -D_XOPEN_SOURCE=600 -I/data/data/com.termux/files/usr/include cdk/selection.c -o cdk-selection.o
gcc -c -Os -D_GNU_SOURCE -Itermkey -Icdk -D_XOPEN_SOURCE=600 -I/data/data/com.termux/files/usr/include cdk/slider.c -o cdk-slider.o
gcc -c -Os -D_GNU_SOURCE -Itermkey -Icdk -D_XOPEN_SOURCE=600 -I/data/data/com.termux/files/usr/include cdk/traverse.c -o cdk-traverse.o
gcc -c -Os -D_GNU_SOURCE -Itermkey -Icdk -D_XOPEN_SOURCE=600 -I/data/data/com.termux/files/usr/include cdk/version.c -o cdk-version.o
g++ -Os -std=c++17 -o ../textadept-curses AutoComplete.o CallTip.o CaseConvert.o CaseFolder.o CellBuffer.o CharacterCategoryMap.o CharacterType.o CharClassify.o ContractionState.o DBCS.o Decoration.o Document.o EditModel.o Editor.o EditView.o Geometry.o Indicator.o KeyMap.o LineMarker.o MarginView.o PerLine.o PositionCache.o RESearch.o RunStyles.o ScintillaBase.o Selection.o Style.o UniConversion.o UniqueString.o ViewStyle.o XPM.o Accessor.o CharacterCategory.o CharacterSet.o DefaultLexer.o LexerBase.o LexerModule.o LexerNoExceptions.o LexerSimple.o PropSetSimple.o StyleContext.o WordList.o LexLPeg-curses.o ScintillaCurses.o textadept-curses.o lapi.o lcode.o lctype.o ldebug.o ldo.o ldump.o lfunc.o lgc.o linit.o llex.o lmem.o lobject.o lopcodes.o lparser.o lstate.o lstring.o ltable.o ltm.o lundump.o lvm.o lzio.o lauxlib.o lbaselib.o lcorolib.o ldblib.o liolib.o lmathlib.o loadlib.o lstrlib.o ltablib.o lutf8lib.o loslib-curses.o lpcap.o lpcode.o lpprint.o lptree.o lpvm.o lfs.o gtdialog-curses.o termkey.o driver-ti.o driver-csi.o cdk-binding.o cdk-buttonbox.o cdk-cdk.o cdk-cdk_display.o cdk-cdk_objs.o cdk-cdkscreen.o cdk-draw.o cdk-entry.o cdk-fselect.o cdk-itemlist.o cdk-label.o cdk-mentry.o cdk-popup_label.o cdk-scroll.o cdk-scroller.o cdk-select_file.o cdk-selection.o cdk-slider.o cdk-traverse.o cdk-version.o -L/data/data/com.termux/files/usr/lib -Wl,-rpath=/data/data/com.termux/files/usr/lib -fopenmp -static-openmp -Wl,--enable-new-dtags -Wl,--as-needed -lncursesw -rdynamic -Wl,--retain-symbols-file -Wl,/data/data/com.termux/files/home/downloads/textadept/src/lua.sym -ldl
/data/data/com.termux/files/usr/bin/ld: textadept-curses.o: in function `iconv_lua':
textadept.c:(.text+0x3554): undefined reference to `libiconv_open'
/data/data/com.termux/files/usr/bin/ld: textadept.c:(.text+0x35ac): undefined reference to `libiconv'
/data/data/com.termux/files/usr/bin/ld: textadept.c:(.text+0x3620): undefined reference to `libiconv_close'
/data/data/com.termux/files/usr/bin/ld: textadept.c:(.text+0x3644): undefined reference to `libiconv'
/data/data/com.termux/files/usr/bin/ld: textadept.c:(.text+0x3670): undefined reference to `libiconv_close'
clang-12: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [Makefile:207: textadept-curses] Error 1

**Comment by orbitalquark** [comment]
Try changing line 60 *src/Makefile* to "LDFLAGS += -ldl -liconv". For whatever reason, libiconv is not being included on your platform. I'm not sure why that is.

**Reply by kumoshk** [reply]
It worked! :) Thanks.
